coorx/linear.py

```diff
@@ -38,26 +38,84 @@ def _imap(self, coords):
         return coords
 
     def as_affine(self):
-        return AffineTransform(matrix=np.eye(self.dims[0]), offset=np.zeros(self.dims[0]))
+        return AffineTransform(
+            matrix=np.eye(self.dims[0]), offset=np.zeros(self.dims[0]), from_cs=self.systems[0], to_cs=self.systems[1]
+        )
 
     @property
     def full_matrix(self):
-        return np.eye(self.dims[0]+1)
+        return np.eye(self.dims[0] + 1)
 
     def __mul__(self, tr):
-        return tr
+        from coorx import CompositeTransform
+
+        if isinstance(tr, CompositeTransform):
+            return tr.__rmul__(self)
+        if isinstance(tr.inverse, CompositeTransform):
+            return tr.inverse.__mul__(self.inverse).inverse
+        self.validate_transform_for_mul(tr)
+        return tr.copy(from_cs=tr.systems[0], to_cs=self.systems[1])
 
     def __rmul__(self, tr):
-        return tr
+        from coorx import CompositeTransform
+
+        if isinstance(tr, CompositeTransform):
+            return tr.__mul__(self)
+        if isinstance(tr.inverse, CompositeTransform):
+            return tr.inverse.__rmul__(self.inverse).inverse
+        tr.validate_transform_for_mul(self)
+        return tr.copy(from_cs=self.systems[0], to_cs=tr.systems[1])
 
     @property
     def params(self):
         return {}
-    
+
     def set_params(self):
         return
 
 
+class TransposeTransform(Transform):
+    """Transposes axes of input coordinates (as opposed to dimensions)."""
+
+    Linear = True
+    Orthogonal = True
+    NonScaling = True
+    Isometric = False
+    state_keys = ["axes"]
+
+    def __init__(self, axes: tuple[int, ...] = None, *args, **kwargs):
+        if "dims" not in kwargs:
+            kwargs["dims"] = (len(axes),) * 2
+        super().__init__(*args, **kwargs)
+        self.axes = axes
+
+    def _map(self, coords):
+        """Return the input array with columns swapped."""
+        return coords[:, self.axes]
+
+    def _imap(self, coords):
+        """Return the input array columns inversely swapped."""
+        return coords[:, np.argsort(self.axes)]
```

Comment:
cache the inverse axis order?

Reply:
cache invalidation when axis_order is changed deemed too much complexity for now.